Добавлен параметр для обновления расширения из хранилища

Borisskin opened the pull request:
Не хватало параметра вызова, который уже есть в МенеджерКонфигуратора.ЗапуститьОбновлениеИзХранилища
При вызове runner loadrepo в cmd добавить параметр вызова, где NameExtension - имя расширения 
`call runner loadrepo --storage-ext "-Extension \"NameExtension\""`

Borisskin committed:
Обновление main.os для работы с расширением

Добавлен "--storage-ext", который будет использован для доп.ключей запуска обновления из хранилища

Borisskin committed:
Обновление КомандаОбновитьИзХранилища.os

Добавлен "--storage-ext", который используется как доп.ключи запуска обновления из хранилища. Для возможности обновить расширение.

artbear commented on src/Классы/КомандаОбновитьИзХранилища.os:
```diff
@@ -31,6 +31,7 @@
 	Парсер.ДобавитьИменованныйПараметрКоманды(ОписаниеКоманды, "--storage-pwd", "Пароль");
 	Парсер.ДобавитьИменованныйПараметрКоманды(ОписаниеКоманды, "--storage-ver",	
 		"Номер версии, по умолчанию берем последнюю");
+	Парсер.ДобавитьИменованныйПараметрКоманды(ОписаниеКоманды, "--storage-ext",	"Доп. ключи запуска");
```
Предлагаю еще добавить спец. параметр "--extension" для указания имени расширения

и прокидывать его, передавая как еще один доп.параметр.

т.е. будет 2 новых параметра у команды.

в итоге код будет такой
```
ДопПараметры = СтрШаблон("%1 %2", ПараметрыКоманды["--extension"], ПараметрыКоманды["--storage-ext"]);
МенеджерКонфигуратора.ЗапуститьОбновлениеИзХранилища(
			ПараметрыКоманды["--storage-name"], ПараметрыКоманды["--storage-user"],
			ПараметрыКоманды["--storage-pwd"], 
			ПараметрыКоманды["--storage-ver"]);
			ПараметрыКоманды["--storage-ver"], ДопПараметры);
```

artbear replied:
И получим простую команду
`call runner loadrepo --extension "NameExtension"`

Borisskin replied:
Чтобы сработало указание на расширение, необходимо в строке запуска из cmd добавить
-Extension "NameExtension"

Предлагаемый вариант с параметром --extension ведёт к тому, что надо дополнительно изменить код в МенеджерКонфигуратора.ЗапуститьОбновлениеИзХранилища и там обрабатывать параметр, и если он не пусто добавлять к строке запуска "-Extension"
Или я не так понял предлагаемый вариант?

Borisskin replied:
Сейчас столкнулся с тем, что в КомандаОтключитьсяОтХранилища.os тоже не хватает параметра для указания расширения, как и для КомандаПодключитьсяКХранилищу.os.
И теперь согласен с тем, что надо вводить просто параметр для имени расширения.

Почему-то в других *.os параметр имени расширения объявлен как 'extensionName', без двойного дефиса в начале. Привести бы к общему объявлению параметра.

artbear reviewed (changes requested):
немного бы дополнить

Borisskin committed:
Обновление МенеджерКонфигуратора.os

Изменен код ПодключитьсяКХранилищу - чтобы можно было использовать дополнительные ключи запуска
Изменен код ОтключитьсяОтХранилища - проверка на дополнительные ключи запуска и изменение текста лога

Borisskin committed:
Обновление КомандаОбновитьИзХранилища.os

Введен именованный параметр для имени хранилища

Borisskin committed:
Обновление КомандаПодключитьсяКХранилищу.os

Изменен код для использования расширения

Borisskin committed:
Обновление КомандаОтключитьсяОтХранилища.os

Добавлен параметр команды для передачи имени расширения

Borisskin committed:
Обновление main.os

Замена переменной для имени расширения --extension

Borisskin committed:
Update КомандаОбновитьИзХранилища.os

Borisskin committed:
Update КомандаОтключитьсяОтХранилища.os

Borisskin committed:
Update КомандаПодключитьсяКХранилищу.os

Borisskin committed:
Update МенеджерКонфигуратора.os

artbear commented on src/Классы/КомандаПодключитьсяКХранилищу.os:
```diff
@@ -25,18 +25,17 @@
-	Парсер.ДобавитьПозиционныйПараметрКоманды(ОписаниеКоманды, "Пароль", "Пароль пользователя хранилища 1С");
-	
+
+	Парсер.ДобавитьИменованныйПараметрКоманды(ОписаниеКоманды, "--storage-name", "Строка подключения к хранилищу");
```
вот эти изменения поломают существующие команды и CI (

т.к. уходим от позиционных параметров к указанию имен.

artbear replied:
пока предлагаю параметры логина и пароля оставить позиционными, а имя расширения указать как именованный параметр.

artbear commented on src/Классы/МенеджерКонфигуратора.os:
```diff
@@ -551,15 +567,17 @@
-	Параметры.Добавить("/ConfigurationRepositoryUnbindCfg"); 
-	Параметры.Добавить("-force");
+	Параметры.Добавить("/ConfigurationRepositoryUnbindCfg "); 
+	Параметры.Добавить("-force ");
```
зачем нужны доп.пробелы в конце 2х ключей ??
- "/ConfigurationRepositoryUnbindCfg "
- "-force "

Borisskin replied:
Тут перемудрил и пробелы, действительно, лишние

Borisskin replied:
Почему в МенеджерКонфигуратора.ПодключитьсяКХранилищу для подключения используется МенеджерХранилищаКонфигурации, а не сразу УправлениеКонфигуратором?
Первый не принимает имя расширения, его дописывать надо. Во второй же просто добавить параметр на выполнение.
Как раз из-за CI?

artbear commented on src/Классы/МенеджерКонфигуратора.os:
```diff
@@ -551,15 +567,17 @@
 	
 	Если Не ПустаяСтрока(ДополнительныеКлючиЗапуска) Тогда
 		Параметры.Добавить(ДополнительныеКлючиЗапуска);
+		Лог.Информация("Выполняю отключение от хранилища расширения");
```
считаю неверной проверку на хранилище расширения выполнять через 
`Если Не ПустаяСтрока(ДополнительныеКлючиЗапуска)`

закладываем мину на будущее.
если в будущем добавятся другие доп.ключи, это условие будет неверным.

artbear commented on src/Классы/МенеджерКонфигуратора.os:
```diff
@@ -502,36 +502,52 @@
+	
+	Если Не ПустаяСтрока(ДополнительныеКлючиЗапуска) Тогда
+		Параметры.Добавить(ДополнительныеКлючиЗапуска);
+		Лог.Информация("Выполняю подключение к хранилищу расширения");
```
и здесь считаю неверной проверку на хранилище расширения выполнять через 
`Если Не ПустаяСтрока(ДополнительныеКлючиЗапуска)`

закладываем мину на будущее.
если в будущем добавятся другие доп.ключи, это условие будет неверным.

artbear commented on src/Классы/КомандаПодключитьсяКХранилищу.os:
```diff
@@ -67,12 +66,17 @@
+	ДопПараметры = "";
+	ИмяРасширения = ПараметрыКоманды["--extension"];
+	Если Не ПустаяСтрока(ИмяРасширения) Тогда
+		ДопПараметры = "-Extension """+ИмяРасширения+"""";
```
использование спрятанных доп.параметров в виде строки считаю нехорошим кодом для передачи в АПИ-функции МенеджераКонфигуратора.

Предлагаю к методам
- МенеджерКонфигуратора.ПодключитьсяКХранилищу
- МенеджерКонфигуратора.ОтключитьсяОтХранилища
либо
- добавить по одному параметру ИмяРасширения
- либо сделать спец.методы - **это лучшее**
- МенеджерКонфигуратора.ПодключитьсяКХранилищуРасширения
- МенеджерКонфигуратора.ОтключитьсяОтХранилищаРасширения

это же АПИ, его удобнее разделять.

Borisskin replied:
Т.к. не сильно разбираюсь, то и не понимаю полезности разделения методов, т.к. из-за одного параметра добавленного, потом поддерживать надо уже 2 метода одного направления. 
Пока (лично) выбираю добавление параметра ИмяРасширения к методам:
- МенеджерКонфигуратора.ОтключитьсяОтХранилища
- МенеджерКонфигуратора.ПодключитьсяКХранилищу
- МенеджерКонфигуратора.ЗапуститьОбновлениеИзХранилища

Либо, когда в озвученных командах встречает параметр ИмяРасширения, то можно вызывать в МенеджерКонфигуратора метод для Расширения. Ты это имел в виду?

artbear reviewed (changes requested):
есть много вопросов для обсуждения.
Обсудим/исправим?

artbear commented:
В общем, давай обсудим, исправим и включим в следующий релиз.

в релиз 1.9.0 не вошло из-за вопросов (

artbear commented:
@Borisskin обсудим?

Borisskin commented:
Привет. Извини, загруженные дни были. Сегодня вечером посмотрю и отвечу.

вт, 17 дек. 2019 г., 8:56 Artur Ayukhanov <notifications@github.com>:

> @Borisskin <https://github.com/Borisskin> обсудим?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/silverbulleters/vanessa-runner/pull/310?email_source=notifications&email_token=AAXXCTOMINROAR6YDVDH7Y3QZBSZXA5CNFSM4ISJJN22YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEHBHYBI#issuecomment-566393861>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAXXCTPMYC3WIY3FRGZ4J4DQZBSZXANCNFSM4ISJJN2Q>
> .
>

Smilebringer commented:
Привет, еще придется модифицировать команду обновления БД. Ловлю такой момент, что при вызове updatedb не обновляет расширение. Думаю связано с тем, что без явного указания имени расширения обновляется только основная конфигурация

Smilebringer commented:
Отбой, поторопился, платформенная бага
[https://bugboard.v8.1c.ru/error/000046991](url)

Borisskin committed:
Update МенеджерКонфигуратора.os

Изменение команд под параметр ИмяРасширения

Borisskin committed:
Update КомандаОбновитьИзХранилища.os

Добавление параметра ИмяРасширения в вызов команды. Убрал из доппараметров

Borisskin committed:
Update КомандаОтключитьсяОтХранилища.os

Добавление параметра ИмяРасширения в вызов команды. Убрал из доппараметров

Borisskin committed:
Update КомандаПодключитьсяКХранилищу.os

Добавление параметра ИмяРасширения в вызов команды. Убрал из доппараметров.

Borisskin committed:
Update МенеджерКонфигуратора.os

Borisskin committed:
Update МенеджерКонфигуратора.os

PavelDv commented:
МенеджерКонфигуратора.os, строка 559
Переменная логинАдминистратора не объявлена. Просто Логин.

Borisskin committed:
Исправление ошибки

Старый код в 559. Вместо Логин было ЛогинАдминистратора

artbear commented:
@Borisskin сейчас в твоем ПР возник конфликт кода.

1 исправишь?
2 можешь добавить простое минимальное хранилище расширения
  - в нем 2 коммита для расширения
  - расширение простейшее для базы Ванесса-АДД (83Sync) - добавлен новый отчет, например

а я бы на основании этого расширения сделал бы фичи для проверки твоего кода

хочется включить твой ПР в текущий релиз, как и обещал )
очень жду.

Borisskin committed:
попытка испрвления конфликта кода в ПР

взял последний из develop и добавил свой код

Borisskin committed:
Merge branch 'develop' of https://github.com/silverbulleters/vanessa-runner into silverbulleters-develop

Borisskin committed:
Merge branch 'silverbulleters-develop' into develop

Borisskin committed:
Добавлено хранилище расширения для базы Ванесса-АДД (83Sync)

Borisskin committed:
Отключил от хранилища расширение

Borisskin committed:
Описание хранилища (пользователь-пароль)

Borisskin committed:
Удалены временные cfl-файлы

Borisskin commented:
На этой неделе сильно занят, если не получится, то на следующей точно сделаю.

Borisskin committed:
Расширение в создании хранилища

Borisskin committed:
Восстановил параметр расширения

Borisskin commented:
Автоматом коммит сюда, а не в #391.
Не сверял, но ключей для параметра расширения можно много реализовать: [обсуждение на инфостарт](http://forum.infostart.ru/forum24/topic234969/message2381166/#message2381166)

Borisskin committed:
Merge branch 'develop' into develop

Borisskin committed:
Merge branch 'vanessa-opensource:develop' into develop

Segate-ekb commented:
@Borisskin  Пр еще актуален? Или можно его закрыть? ) 
последнее изменение 3 года назад...

Borisskin commented:
> @Borisskin  Пр еще актуален? Или можно его закрыть? ) 
> последнее изменение 3 года назад...

Я думаю можно закрыть. Первые комментарии были 5 лет назад. Оно даже в дев не попадало. А где я это использовал я там полгода уже не работаю.
